app/controllers/carriers_controller.rb

```diff
@@ -0,0 +1,43 @@
+require 'timeout'
+
+class CarriersController < ApplicationController
+  def index
+    ### sample params ###
+    # {"origin"=>{:country=>"US", :state=>"CA", :city=>"Beverly Hills", :zip=>"90210"},"destination"=>{:country=>"US", :state=>"WA", :city=>"Seattle", :zip=>"98103"},"packages"=>[{:weight=>100, :height=>50, :length=>20, :width=>30}]}
+
+    # create ActiveShipping Locations and Packages
+    
+    # origin = Carrier.create_origin(params[:origin])
+    # destination = Carrier.create_destination(params[:destination])
+    # packages = Carrier.create_packages(params[:packages]) # as an array
+    #
+    #
+    # @ups_rates = begin
+    #   Carrier.get_rates(Carrier.activate_ups, origin, destination, packages)
+    # rescue Timeout::Error
+    #   nil
+    # end
+    #
+    # @usps_rates = begin
+    #   Carrier.get_rates(Carrier.activate_usps, origin, destination, packages)
+    # rescue Timeout::Error
+    #   nil
+    # end
```

Comment:
I noticed this commented-out code got moved into the carriers model. It would have been good to have removed the code from here, although I totally understand that this was a week-long project and you may not have had time to refactor and clean everything.